scripts/dts/edtlib.py

```diff
@@ -773,7 +773,17 @@ def _init_binding(self):
                 # specified bus (if any) and then against any bus. This is so
                 # that matching against bindings which do not specify a bus
                 # works the same way in Zephyr as it does elsewhere.
-                if (compat, on_bus) in self.edt._compat2binding:
+                if isinstance(on_bus, list) \
+                    and ((compat, elem) in self.edt._compat2binding for elem in on_bus):
+                    for elem in on_bus:
+                        if (compat, elem) in self.edt._compat2binding:
+                            # When matching, the preference is encoded in to
+                            # bus list order, and it is to match the first one
+                            # available
+                            binding = self.edt._compat2binding[compat, elem]
+                            break
+                elif isinstance(on_bus, str) \
```

Comment:
I think the isinstance test can be dropped here, no?

Reply:
It can't...

It's possible that the above if statement could fail even if it is list and it doesn't find it in the self.edt._compat2binding.

It will then attempt to do `(compat, on_bus) in self.edt._compat2binding` and will fail due to `on_bus` being a list.